packages/core-v2/src/visual-model/visual-model.ts

```diff
@@ -1,183 +1,138 @@
-
-export interface VisualEntityModel {
-    getId(): string;
-    getVisualEntity(entityId: string): VisualEntity | undefined;
-    getVisualEntities(): Map<string, VisualEntity>;
-    addEntity(entity: Partial<Omit<VisualEntity, "id" | "type">>): void;
-    updateEntity(
-        visualEntityId: string,
-        entity: Partial<Omit<VisualEntity, "id" | "type" | "sourceEntityId">>
-    ): boolean;
-    subscribeToChanges(callback: (updated: Record<string, VisualEntity>, removed: string[]) => void): () => void;
-    deserializeModel(data: object): VisualEntityModel;
-
-    getColor(semModelId: string): string | undefined;
-    setColor(semModelId: string, hexColor: string): void;
-
-    /** [modelId: string, hexColor: string] */
-    getModelColorPairs(): [string, string][];
+import { EntityIdentifier } from "./entity-model/entity";
+import { EntityModel } from "./entity-model/entity-model";
+import { LegacyModel } from "./entity-model/legacy-model";
+import { ObservableEntityModel, UnsubscribeCallback } from "./entity-model/observable-entity-model";
+import { SynchronousEntityModel } from "./entity-model/synchronous-entity-model";
+import { SynchronousWritableEntityModel } from "./entity-model/on-premise-writable-entity-model";
+import { isTypedObject } from "./entity-model/typed-object";
+import { VisualEntity, VisualGroup, VisualNode, VisualRelationship } from "./visual-entity";
+import { SerializableModel } from "./entity-model/serializable-model";
+
+export type RepresentedEntityIdentifier = string;
+
+/**
+ * Color stored in hex with the leading hash sign, e.g. '#70d6ff'.
+ */
+export type HexColor = string;
+
+/**
+ * Visual model is designed to allow users place a class, or profile, on a canvas.
+ * This include, but is not limited to, associating position and color with an entity.
+ *
+ * Since the visual model capture what use see we design it as synchronous interface.
+ */
+export interface VisualModel extends LegacyModel {
+
+    /**
+     * @returns Model identifier.
+     */
+    getIdentifier(): string;
+
+    /**
+     * @returns Visual entity with given identifier or null.
+     */
+    getVisualEntity(identifier: EntityIdentifier): VisualEntity | null;
+
+    /**
+     * @returns Entity with given source entity identifier or null.
+     */
+    getVisualEntityForRepresented(represented: RepresentedEntityIdentifier): VisualEntity | null;
+
+    /**
+     * @returns Snapshot of map with all entities in the model.
+     */
+    getVisualEntities(): Map<EntityIdentifier, VisualEntity>;
+
+    /**
+     * Subscribe to changes.
+     * @returns Callback to cancel the subscription.
+     */
+    subscribeToChanges(listener: VisualModelListener): UnsubscribeCallback;
+
+    /**
+     * @returns Color as defined for given model or null.
+     */
+    getModelColor(identifier: string): HexColor | null;
+
+    /**
+     * @returns All stored model color pairs.
+     */
+    getModelColors(): Map<string, VisualModelData>;
```

Comment:
Should this method return the whole `VisualModelData` or just the color (i. e. `HexColor`). To me it makes sense that only the color, but I am not sure what will be the future content of the `VisualModelData` interface.